JS/Lesson19/goksel/recursion.js

```diff
@@ -36,103 +36,231 @@ console.log('5', getFactorial(5));
 const sentenceEx5 = 'Hello, how are you?';
 const charToRemove = 'o';
 
+function removeCharacter(sentence, removeWord){
+    if(sentence.length === 0 || sentence[0].length === 0){
+        return "";
+        }
+        if(sentence[0] === removeWord ){
+            return removeCharacter(sentence.slice(1), removeWord)
+        }else {
+            return sentence[0] + removeCharacter(sentence.slice(1), removeWord);
+        }
+}
+
+console.log(removeCharacter(sentenceEx5,charToRemove));
+
+console.log("---Ex5---");
 // 5. Write a recursive function to check if an array includes a specific value.
 const numbers = [1, 2, 3, 4, 5];
-const valueToCheck = 3;
+const valueToCheck = 2;
+
+
+function isArrayIncludeValue(array, value){
+    if(array.length === 0){
+        return "Value is not found";
+    } 
+    if(array[0] !== value){
+        return isArrayIncludeValue(array.slice(1), value);
+    } else{
+        return true;
+    }
+
+}
 
+console.log(isArrayIncludeValue(numbers, valueToCheck));
+console.log("---Ex6---");
 // 6. Write a recursive function to flatten an object with nested objects into a single-level object.
 const nestedObj = {
-	name: 'John',
-	age: 30,
-	address: {
-		street: '123 Main St',
-		city: 'New York',
-		country: 'USA',
-	},
-	hobbies: ['reading', 'coding'],
-	social: {
-		twitter: '@johndoe',
-		linkedIn: 'johndoe',
-	},
+    name: 'John',
+    age: 30,
+    address: {
+        street: '123 Main St',
+        city: 'New York',
+        country: 'USA',
+    },
+    hobbies: ['reading', 'coding'],
+    social: {
+        twitter: '@johndoe',
+        linkedIn: 'johndoe',
+    },
 };
 
+
+
+// I have no idea about this example...
+
+console.log("---Ex7---");
 // 7. Write a recursive function to find the maximum depth of a nested object.
 const nestedObjEx7 = {
-	a: 1,
-	b: {
-		c: 2,
-		d: {
-			e: 3,
-			f: {
-				g: 4,
-			},
-		},
-	},
-	h: 5,
+    a: 1,
+    b: {
+        c: 2,
+        d: {
+            e: 3,
+            f: {
+                g: 4,
+            },
+        },
+    },
+    h: 5,
 };
+
+
+function maxDepthOfObject(obj, depth = 0) {
+    if (typeof obj !== "object" || obj === null){
+        return depth;
+    }
+
+    let maxDepth = depth;
+    for(let key in obj){
+        if(obj.hasOwnProperty(key)){
+            let currentDepth = maxDepthOfObject(obj[key], depth +1);
+            if (currentDepth > maxDepth){
+                maxDepth = currentDepth;
+            }
+        }
+    }
+    return maxDepth;
+}
+
+console.log(maxDepthOfObject(nestedObjEx7));
+console.log("---Ex8---");
+
+// I checked this solution from chatgbt, ı tried to understand but not perfect.
+
 // 8. Write a recursive function to reverse the order of words in a sentence.
 const sentenceEx8 = 'Hello, how are you?';
+
+
+function reverseOrder(sentence){
+    let splitArray = sentence.split(" "); 
+    
+    if(splitArray.length === 0 || splitArray[0].length === 0){
```

Comment:
Condition `splitArray[0].length === 0` makes sense here  we split the initial string into array on the line above

Reply:
yes ı understand that better now.